src/main/java/htsjdk/samtools/QueryInterval.java

```diff
@@ -94,4 +94,24 @@ public String toString() {
 
         return unique.toArray(EMPTY_QUERY_INTERVAL_ARRAY);
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
In classes that implement `Comparable` it's preferable to implement `equals()` as `this.compareTo(that) == 0` after the initial type checks.

Reply:
@tfenne, done. Thank you for your note.